src/components/shared/TractionBadge.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <span :class="`badge badge-${color}`">
+    <slot />
+  </span>
+</template>
+
+<script>
+export default {
+  /**
+   * # TractionBadge
+   *
+   * Tailwind component to display a coloured badge with text.
+   * The colour can be set using the `color` prop.
+   * Available colours are: `gray`, `red`, `yellow`, `green`, `blue`, `indigo`, `purple`, `pink`.
+   * The default colour is `blue`.
+   *
+   * @example
+   * <template>
+   *   <div>
+   *      <traction-badge>
+   *        Badge Text
+   *      </traction-badge>
+   *   </div>
+   * </template>
+   */
+  name: 'TractionBadge',
+  props: {
+    color: {
+      type: String,
+      default: 'blue',
+      required: false,
+    },
+  },
+}
+</script>
+<style scoped>
+/* badges */
+.badge {
+  @apply inline-flex items-center rounded-lg px-1 py-0 my-1 text-xs font-medium;
```

Comment:
I agree , it is better to avoid mixing Tailwind CSS with inline styles if possible to help maintainability and to maintain consistency across project.